backend/dal/Services/Admin/Concrete/UserService.cs

```diff
@@ -31,18 +31,48 @@ public class UserService : BaseService<User>, IUserService
         #region Methods
         /// <summary>
         /// Get a page of users from the datasource.
+        /// The filter will allow queries to search for anything that starts with the following properties; DisplayName, FirstName, LastName, Email, Agencies.
         /// </summary>
         /// <param name="page"></param>
         /// <param name="quantity"></param>
-        /// <param name="sort"></param>
         /// <returns></returns>
-        public Paged<User> GetNoTracking(int page = 1, int quantity = 10, string sort = null)
+        public Paged<User> GetNoTracking(int page = 1, int quantity = 10)
+        {
+            return GetNoTracking(new UserFilter(page, quantity));
+        }
+
+        /// <summary>
+        /// Get a page of users from the datasource.
+        /// The filter will allow queries to search for the following property values; DisplayName, FirstName, LastName, Email, Agencies.
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public Paged<User> GetNoTracking(UserFilter filter = null)
         {
             this.User.ThrowIfNotAuthorized("system-administrator");
 
+            if (filter.Page < 1) filter.Page = 1;
+            if (filter.Quantity < 1) filter.Quantity = 1;
+            if (filter.Quantity > 50) filter.Quantity = 50;
+
             var query = this.Context.Users.AsNoTracking();
-            var users = query.Skip((page - 1) * quantity).Take(quantity);
-            return new Paged<User>(users.ToArray(), page, quantity, query.Count());
+            if (filter != null)
+            {
+                if (!string.IsNullOrWhiteSpace(filter.DisplayName))
```

Comment:
This approach looks good but doesn't seem to scale well with large numbers of fields. Thoughts?

Reply:
Definitely it's primary flaw.  While it allows us control over what is queryable, it makes us have to work harder for complex options.  I can rewrite using reflection at some point, but for now it doesn't make sense to do this.